lib/HLSL/DxilNoops.cpp

```diff
@@ -582,6 +583,93 @@ Pass *llvm::createDxilRewriteOutputArgDebugInfoPass() {
 
 INITIALIZE_PASS(DxilRewriteOutputArgDebugInfo, "dxil-rewrite-output-arg-debug-info", "Dxil Rewrite Output Arg Debug Info", false, false)
 
+//==========================================================
+// Reader pass
+//
+
+namespace {
+
+class DxilReinsertNops : public ModulePass {
+public:
+  static char ID;
+
+  DxilReinsertNops() : ModulePass(ID) {
+    initializeDxilReinsertNopsPass(*PassRegistry::getPassRegistry());
+  }
+
+  // In various linking scenarios, the dx.nothing.a variable might be prefixed
+  // and/or suffixed with something:
+  //
+  //   <library_name>.dx.nothing.a.<another_thing>
```

Comment:
We don't :(  I actually didn't anticipate linking to add prefixes and suffixes like this. Another option is to have the linker not duplicate these variables (and therefore no hacky string searching, just one strcmp). I think using global variable name is fine, since it's not quite as transient and unreliable as local value names.

Reply:
Ok, so it looks like the expectation is that we can find all the `dx.noop` values by looking for a global with the `hlsl::kNothingName`. In that case I think this pass seems fine.

It also seems like the global variable is more of a "fallback" path for lower shader models. Seems like it would be good to add official support for `dx.noop` to dxil.